metals/src/main/scala/scala/meta/internal/metals/CodeActionProvider.scala

```diff
@@ -46,4 +50,14 @@ final class CodeActionProvider(
     Future.sequence(actions).map(_.flatten)
   }
 
+  def executeCommands(
+      codeActionCommandData: CodeActionCommandData,
+      token: CancelToken
+  )(implicit ec: ExecutionContext): Future[CodeActionCommandResult] = {
+    codeActionCommandData match {
+      case data: ExtractMemberDefinitionData =>
+        extractMemberAction.executeCommand(data)
+      case _ => Future.failed(new IllegalArgumentException)
```

Comment:
```suggestion
      case data => Future.failed(new IllegalArgumentException(data.toString))
```
?

Otherwise we'll need to dig in into what was sent.